pkg/providers/instance/resperrorhandlers_test.go

```diff
@@ -87,12 +87,8 @@ func newTestCase(name string) *testCaseBuilder {
 	}
 }
 
-func (b *testCaseBuilder) withInstanceType(offerings ...offering) *testCaseBuilder {
-	offeringTypes := make([]struct{ zone, capacityType string }, len(offerings))
-	for i, o := range offerings {
-		offeringTypes[i] = struct{ zone, capacityType string }{zone: o.zone, capacityType: o.capacityType}
-	}
-	b.tc.instanceType = defaultCreateInstanceType(offeringTypes...)
+func (b *testCaseBuilder) withDefaultInstanceType(offerings ...offering) *testCaseBuilder {
```

Comment:
minor: Why `defaultInstanceType` vs `instanceType`? What's the difference between a "default type" and "the instance type"?

Reply:
Initially there were tests that used different instance type names, right now they all use 
```
responseErrorTestInstanceName  = "Standard_D2s_v3"
```

will change to 

`func (b *testCaseBuilder) withInstanceType(instanceTypeName string, offerings ...offering) *testCaseBuilder `